app/assets/javascripts/select-kit/addon/components/category-selector.js

```diff
@@ -27,27 +26,70 @@ export default MultiSelectComponent.extend({
     if (!this.blockedCategories) {
       this.set("blockedCategories", []);
     }
-  },
 
-  content: computed("categories.[]", "blockedCategories.[]", function () {
-    const blockedCategories = makeArray(this.blockedCategories);
-    return Category.list().filter((category) => {
-      if (category.isUncategorizedCategory) {
-        if (this.options?.allowUncategorized !== undefined) {
-          return this.options.allowUncategorized;
+    if (this.siteSettings.lazy_load_categories) {
+      if (this.categoryIds) {
+        if (!Category.hasAsyncFoundAll(this.categoryIds)) {
+          Category.asyncFindByIds(this.categoryIds).then(() => {
+            this.notifyPropertyChange("categoryIds");
+          });
         }
+      } else {
+        // eslint-disable-next-line no-console
+        console.warn(
+          "categoryIds is undefined, but lazy_load_categories is enabled"
+        );
+      }
 
-        return this.selectKit.options.allowUncategorized;
+      if (this.blockedCategoryIds) {
+        if (!Category.hasAsyncFoundAll(this.blockedCategoryIds)) {
+          Category.asyncFindByIds(this.blockedCategoryIds).then(() => {
+            this.notifyPropertyChange("blockedCategoryIds");
+          });
+        }
```

Comment:
This may end up triggering two requests just to initialise the category selectors. If we're using multiple category selectors on the page, will this end up triggering multiple requests in a short span of time? Is there perhaps a way for us to batch things up on a per page basis?

Reply:
Yes, that is true, but that is not a concern right now because I want to make it work first.

Reply:
:+1: